meilisearch-http/src/index_controller/uuid_resolver.rs

```diff
@@ -127,45 +129,45 @@ impl UuidResolverHandle {
         Ok(Self { sender })
     }
 
-    pub async fn resolve(&self, name: String) -> anyhow::Result<Uuid> {
+    pub async fn get(&self, name: String) -> Result<Uuid> {
         let (ret, receiver) = oneshot::channel();
-        let msg = UuidResolveMsg::Resolve { uid: name, ret };
+        let msg = UuidResolveMsg::Get { uid: name, ret };
         let _ = self.sender.send(msg).await;
         Ok(receiver
             .await
             .expect("Uuid resolver actor has been killed")?)
     }
 
-    pub async fn get_or_create(&self, name: String) -> Result<Uuid> {
+    pub async fn create(&self, name: String) -> anyhow::Result<Uuid> {
         let (ret, receiver) = oneshot::channel();
-        let msg = UuidResolveMsg::GetOrCreate { uid: name, ret };
+        let msg = UuidResolveMsg::Create { uid: name, ret };
         let _ = self.sender.send(msg).await;
         Ok(receiver
             .await
             .expect("Uuid resolver actor has been killed")?)
     }
 
-    pub async fn create(&self, name: String) -> anyhow::Result<Uuid> {
+    pub async fn delete(&self, name: String) -> anyhow::Result<Uuid> {
         let (ret, receiver) = oneshot::channel();
-        let msg = UuidResolveMsg::Create { uid: name, ret };
+        let msg = UuidResolveMsg::Delete { uid: name, ret };
         let _ = self.sender.send(msg).await;
         Ok(receiver
             .await
             .expect("Uuid resolver actor has been killed")?)
     }
 
-    pub async fn delete(&self, name: String) -> anyhow::Result<Uuid> {
+    pub async fn list(&self) -> anyhow::Result<Vec<(String, Uuid)>> {
         let (ret, receiver) = oneshot::channel();
-        let msg = UuidResolveMsg::Delete { uid: name, ret };
+        let msg = UuidResolveMsg::List { ret };
         let _ = self.sender.send(msg).await;
         Ok(receiver
             .await
             .expect("Uuid resolver actor has been killed")?)
     }
 
-    pub async fn list(&self) -> anyhow::Result<Vec<(String, Uuid)>> {
+    pub async fn insert(&self, name: String, uuid: Uuid) -> anyhow::Result<()> {
```

Comment:
What is the difference between create and insert?

Reply:
Create generates the uuid for you, but it's likely to disappear in the future